pkg/driver/nodeServer.go

```diff
@@ -300,107 +336,107 @@ func (s *NodeServer) CreateNewTargetTg(params CreateTargetTgParams, nsProvider n
     err = nsProvider.CreateUpdateTargetGroup(createTargetGroupParams)
     if err != nil {
     }
+
+    if parsedContext.UseChapAuth == true {
+        nodeIQN, err := s.GetNodeIQN()
+        if err != nil {
+            return target, targetGroup, err
+        }
+
+        err = s.SetChapAuth(
+            nodeIQN, parsedContext.ChapUser, parsedContext.ChapSecret, nsProvider)
+        if err != nil {
+            return target, targetGroup, err
+        }
+        // Set authentication to CHAP for iSCSI target
+        updateParams := ns.UpdateISCSITargetParams{
+            Authentication: "chap",
+        }
+        err = nsProvider.UpdateISCSITarget(target, updateParams)
+        if err != nil {
+            return target, targetGroup, err
+        }
+    }
+
     return target, targetGroup, err
```

Comment:
Same here and I think there are more similar changes that are not related to the ticket.